### PR TITLE
[2.3] Static Code Analysis for Components

### PR DESCRIPTION
**Pull request by kalessil**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

Static Code Analysis with Php Inspections (EA Extended):
    - not optimal regular expressions usage
    - strlen miss-use
    - not optimal conditional statements in Process and Filesystem
    - unsafe uniquid usage

PS: re-creating to no spam history log with reverts

### CODE REVIEW

**Comment by Tobion** [comment]
:+1:

**Comment by sstok** [comment]
:+1:

File: src/Symfony/Component/Process/Process.php [hunk 1]
```diff
@@ -468,7 +468,7 @@ public function getIncrementalErrorOutput()
      */
     public function getExitCode()
     {
-        if ($this->isSigchildEnabled() && !$this->enhanceSigchildCompatibility) {
+        if (!$this->enhanceSigchildCompatibility && $this->isSigchildEnabled()) {
```
Comment (fabpot) [inline src/Symfony/Component/Process/Process.php]: I'm against all these changes. IIUC, it's a small perf optim, but probably negligible and it makes it very difficult to merge branches between them.

Comment (kalessil) [inline src/Symfony/Component/Process/Process.php]: Didn't take into account emerging issues, but you absolutely right here. Will revert this and according changes in Processes.

**Comment by dunglas** [comment]
:+1:

**Comment by Tobion** [comment]
Thank you @kalessil.

**Pull request closed without merging**